src/main/java/com/google/devtools/build/remote/client/DockerUtil.java

```diff
@@ -28,27 +28,33 @@
   private static final String CONTAINER_IMAGE_ENTRY_NAME = "container-image";
   private static final String DOCKER_IMAGE_PREFIX = "docker://";
 
-  /**
-   * Gets uid of the current user. If the uid could not be fetched, prints a message to stderr and
-   * returns -1.
-   */
   @VisibleForTesting
-  static long getUid() {
-    ProcessBuilder processBuilder = new ProcessBuilder();
-    processBuilder.command("id", "-u");
-    try {
-      InputStream stdout = processBuilder.start().getInputStream();
-      byte[] output = ByteStreams.toByteArray(stdout);
-      return Long.parseLong(new String(output).trim());
-    } catch (IOException | NumberFormatException e) {
-      System.err.printf(
-          "Could not fetch UID for passing to Docker container. The provided docker "
-              + "command will not specify a uid (error: %s)\n",
-          e.toString());
-      return -1;
+  static class UidGetter {
```

Comment:
In general, static variables that are changed by test methods is something to avoid. That introduces dependency on test order.
The more standard pattern for this dependency injection is to have the DockerUtil accept a Provider<long> getUid as a constructor parameter in a VisibleForTesting constructor (with a public default overload). Then you can either construct and keep a DockerUtil instance everywhere, or use the Singleton pattern (given that it's only one method, might as well just have a static wrapper for that method).

Reply:
Makes sense. I don't think there's a need for Singleton, at least for now: it's only being constructed in one place; it's also lightweight so far